src/compiler/moduleNameResolver.ts

```diff
@@ -59,8 +59,7 @@ namespace ts {
 
     interface ModuleResolutionState {
         host: ModuleResolutionHost;
-        // We only use this subset of the compiler options.
-        compilerOptions: { rootDirs?: string[], baseUrl?: string, paths?: MapLike<string[]> };
+        compilerOptions: CompilerOptions;

```

Comment:
Turns out that since `CompilerOptions` consists only of optional properties, anything can be assigned to it. So while I thought I was statically ensuring that these were the only compiler options used, we actually use several others, by passing this into a parameter of type `CompilerOptions` and then accessing them.